### PR TITLE
refactor for cargo vendor

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**
I made some changes with how cargo vendor works. I will update README later. For now, the idea is that users can just use `src`. I replaced `srctar` and `srcdir` with `src`. Source can now determine stuff if it's a directory or a source tarball indicated by `SupportedFormat`

### CODE REVIEW

**Comment by uncomfyhalomacro** [comment]
Working in action. :)

https://github.com/openSUSE/obs-service-cargo_vendor/assets/66054069/320e009e-92b5-4327-b5f2-3eb6a89e1e19

**Comment by uncomfyhalomacro** [comment]
ready now for review.

File: cargo/src/bin/cargo_vendor.rs [hunk 1]
```diff
@@ -82,181 +72,21 @@ fn main() -> Result<(), io::Error> {
...
-        (Some(srcdir), None) => Src::Dir(srcdir.clone()),
-        (None, Some(srctar)) => Src::Tar(srctar.clone()),
-        (Some(_), Some(_)) => {
-            error!("Use only srcdir OR srctar - specifiying both is ambiguous");
-            return Err(io::Error::new(
-                io::ErrorKind::InvalidInput,
-                "Use only srcdir OR srctar",
-            ));
+    match args.src.is_supported() {
+        Ok(kay) => {
```
Comment (Firstyear) [inline cargo/src/bin/cargo_vendor.rs:77]: Missed opportunity for this to be Ok(ay). :P

File: cargo/src/cli/mod.rs [hunk 2]
```diff
@@ -120,3 +120,180 @@ impl Display for Compression {
         write!(f, "{}", msg)
     }
 }
+
+#[derive(clap::Args, Debug, Clone)]
+pub struct Src {
```
Comment (Firstyear) [inline cargo/src/cli/mod.rs:125]: I think this needs (hidden) aliases for SrcTar and SrcDir so that existing _service files "just work" when we upgrade.

Comment (uncomfyhalomacro) [reply cargo/src/cli/mod.rs:125]: True. I will do it in a bit

Comment (Firstyear) [reply cargo/src/cli/mod.rs:125]: No rush :)

**Review by Firstyear — APPROVED** [review]

**Pull request merged**